### ISSUE TITLE
[NEW IDEA] Send a notification when a new owe is created.

### ISSUE THREAD

**Issue by preetjdp** [preetjdp/YouOweMe#15]
**Is your feature request related to a problem? Please describe.**
This a very crucial feature for the entire product.
When a new owe is created many side-effects should take place.

**Describe the idea / solution you'd like**
They being as follows:
- [x] If the user to whom the `Owe` is IssuedTo is present in the system and and has notifications enabled, send a notification to him.
- [x] If the user has the app but does not have notifications enabled, send him an _SMS_  with a deep/dynamic-link url courtesy of [Firebase Dynamic Links](https://firebase.google.com/products/dynamic-links.) with a relevant message. 
- [x] If the user is a first time user (not much details in the sys), then send him a introductory SMS
with the ability to use the app directly from the web as a webApp.
- [x] Add more WIP.

**Additional context**
For all of this have to find something that is super cheap, like super cheap.
Hey might might even consider using the whatsApp api 👀👀

**Comment by preetjdp** [comment]
@pal-codes Am I missing anything?

**Comment by pal-codes** [comment]
Seems good to me.

**Comment by SirusCodes** [comment]
If a user is first time user than they will have the app so, why would we send them sms we can send them notification 🤔

**Comment by pal-codes** [comment]
@SirusCodes 
- Let's say if you add an Owe ie. someone owes you or vice-versa, you need to add his/her name & ph.no (integral for this app.) to let the system know of the ongoing transaction until the person pays back.
- So for the secondary user whose info was fed into the system by You, he/she might need to access the app to check the authenticity of the said/requested transaction or can be any other reason, this is best convenient through the web version of YouOweMe.
- So response to your question, the system keeps a check if a user has surfaced the app at any point, if not -> SMS -> Web App.

**Comment by SirusCodes** [comment]
Ok @pal-codes got it

**Comment by preetjdp** [comment]
All of this is mostly implemented.
Closing for the moment

**Comment by preetjdp** [comment]
Opened this issue back to work on Notification Actions.
Tldr. They Don't work ==> https://github.com/MaikuB/flutter_local_notifications/issues/17

But on further investigation have found two branches of flutter_local_notifications that have support for notifications with actions.
They are 
1. https://github.com/MaikuB/flutter_local_notifications/pull/387
2. https://github.com/MaikuB/flutter_local_notifications/pull/338

Your task should you choose to accept would be to check em out and make something valueable
